src/Microsoft.ApiDesignGuidelines.Analyzers/Core/ExceptionsShouldBePublic.cs

```diff
@@ -24,16 +28,48 @@ public abstract class ExceptionsShouldBePublicAnalyzer : DiagnosticAnalyzer
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Design,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/bb264484.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
+        private static readonly List<string> s_exceptionTypeNames = new List<string>()
+        {
+            "System.Exception",
+            "System.SystemException",
+            "System.ApplicationException"
+        };
+
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterCompilationStartAction(AnalyzeCompilationStart);
+        }
+
+        private void AnalyzeCompilationStart(CompilationStartAnalysisContext csContext)
+        {
+            // Get named type symbols for targetted exception types
+            var exceptionTypes = s_exceptionTypeNames
+                .Select(name => csContext.Compilation.GetTypeByMetadataName(name))
+                .Where(t => t != null)
+                .ToImmutableHashSet();
+
+            // register symbol action for named types
+            csContext.RegisterSymbolAction(saContext => 
+            {
+                var symbol = (INamedTypeSymbol)saContext.Symbol;
+
+                // skip public symbols
+                if (symbol.IsPublic()) return;
+
+                // only report if base type matches 
+                if (exceptionTypes.Contains(symbol.BaseType))
```

Comment:
Current FxCop implementation matches [MSDN doc](https://msdn.microsoft.com/en-us/library/bb264484.aspx), specifically:
> But, if the code has a public exception that later is used as the base for a internal exception, it is reasonable to assume the code further out will be able to do something intelligent with the base exception. The public exception will have more information than what is provided by T:System.Exception, T:System.SystemException, or T:System.ApplicationException.

Even if the public exception is in a different library, it is still likely more specific than `Exception`